### ISSUE TITLE
Warn about mixing logical and ternary operators

### ISSUE THREAD

**Issue by mieszko4** [eslint/eslint#11704]
**What rule do you want to change?**
`no-mixed-operators`

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option with default behaviour to warn

**Please provide some example code that this change will affect:**

```js
// what is priority of this? rule should warn of using brackets explicitly
obj.someValue || conditionSatisfied ? 'good' : 'bad'

// 1 option - rule should not warn
(obj.someValue || conditionSatisfied) ? 'good' : 'bad'

// 2 option - rule should not warn
obj.someValue || (conditionSatisfied ? 'good' : 'bad')
```

**What does the rule currently do for this code?**
it does not warn user about mixed logical and ternary expressions

**What will the rule do after it's changed?**
warn to use brackets around to make it explicit what the priority is

**Are you willing to submit a pull request to implement this change?**
nope, with this issue, I am just checking if this would make sense
Might be related to https://github.com/eslint/eslint/issues/7903

**Comment by mysticatea** [comment]
Thank you for your proposal. I like this idea.

**Comment by platinumazure** [comment]
Hi @mieszko4, thanks for the issue.

Is this not currently possible with `no-mixed-operators`? I'm a little confused.

**Comment by mysticatea** [comment]
The current `no-mixed-operators` checks only binary expressions and logical expressions. This proposal is to add ternary expressions into that.

**Comment by platinumazure** [comment]
Thanks @mysticatea. In that case, I'm :+1: for this enhancement.

**Comment by karthikkp** [comment]
@mysticatea @platinumazure The changes have been made for this and a PR is open for review.
